insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/app_layer/parameter_blocks/TBROverNotificationBlock.kt

```diff
@@ -1,31 +1,20 @@
-package info.nightscout.androidaps.plugins.pump.insight.app_layer.parameter_blocks;
+package info.nightscout.androidaps.plugins.pump.insight.app_layer.parameter_blocks
 
-import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf;
+import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf
 
-public class TBROverNotificationBlock extends ParameterBlock {
+class TBROverNotificationBlock : ParameterBlock() {
 
-    private boolean enabled;
-    private int melody;
-
-    @Override
-    public void parse(ByteBuf byteBuf) {
-        enabled = byteBuf.readBoolean();
-        melody = byteBuf.readUInt16LE();
-    }
-
-    @Override
-    public ByteBuf getData() {
-        ByteBuf byteBuf = new ByteBuf(4);
-        byteBuf.putBoolean(enabled);
-        byteBuf.putUInt16LE(melody);
-        return byteBuf;
-    }
-
-    public boolean isEnabled() {
-        return enabled;
+    var isEnabled = false
+    private var melody = 0
+    override fun parse(byteBuf: ByteBuf) {
+        isEnabled = byteBuf.readBoolean()
+        melody = byteBuf.readUInt16LE()
     }
 
-    public void setEnabled(boolean enabled) {
-        this.enabled = enabled;
+    override fun getData(): ByteBuf {
+        val byteBuf = ByteBuf(4)
+        byteBuf.putBoolean(isEnabled)
+        byteBuf.putUInt16LE(melody)
+        return byteBuf
     }
-}
+}
```

Comment:
kotlin way:
```
override fun getData(): ByteBuf =
        ByteBuf(4).also {
          it.putBoolean(isEnabled)
          it.putUInt16LE(melody)
        }
```
;)

Reply:
Yes but a bit later (parent is still in java...)
The conversion of some classes is very tricky for me because there are a lot of multiple inheritances and if the null management or visibility of each property is not good it breaks a lot of things everywhere. 
So I work very slowly to check on each step if I can still build app and if driver still work...